dicomanonymizer/simpledicomanonymizer.py

```diff
@@ -386,14 +378,27 @@ def anonymize_dataset(dataset: pydicom.Dataset, extra_anonymization_rules: dict
     private_tags = []
 
     for tag, action in current_anonymization_actions.items():
-        action(dataset, tag)
-        try:
-            element = dataset.get(tag)
-        except:
-            print("Cannot get element from tag: ", tag_to_hex_strings(tag))
 
-        if element and element.tag.is_private:
-            private_tags.append(get_private_tag(dataset, tag))
+        def range_callback(dataset, data_element):
+            if data_element.tag.group & tag[2] == tag[0] and data_element.tag.element & tag[3] == tag[1]:
+                action(dataset, tag)
+
+        element = None
+
+        # We are in a repeating group
+        if len(tag) > 2:
+            dataset.walk(range_callback)
+        # Individual Tags
+        else:
+            action(dataset, tag)
+            try:
+                element = dataset.get(tag)
+            except:
+                print("Cannot get element from tag: ", tag_to_hex_strings(tag))
+
+            # Get private tag to restore it later
+            if element and element.tag.is_private:
+                private_tags.append(get_private_tag(dataset, tag))
```

Comment:
shouldn't you also do this in the "range" / "repeating-group" case ?

Reply:
I didn't see any information about repeating group in private tag.
Moreover, the syntax we use int he project is proper to the project and should be encountered in private tags.

Reply:
> I didn't see any information about repeating group in private tag.

Fair enough, but did you see information that repeating groups can't exist in private tags ?

>  Moreover, the syntax we use int he project is proper to the project and should be encountered in private tags.

I'm not sure to understand what you mean here

Reply:
It exists only two repeating groups which are not used in private tags : http://dicom.nema.org/dicom/2013/output/chtml/part05/sect_7.6.html

Reply:
Understood.

This link is a good candidate to get into the commit message

Reply:
Done !